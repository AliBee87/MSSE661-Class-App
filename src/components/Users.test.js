import React from 'react';
import { Adapter, shallow, configure } from 'enzyme';
import { UsersTest } from "./Users";

configure({adapter: new Adapter()});

describe("UsersTest", () => {

  it("should render initial layout", () => {
    const component = shallow(<UsersTest />);
    expect(component.getElements()).toMatchSnapshot();

  });
  
  it("should create an entry in component state", () => {
    const component = shallow(<UsersTest />);
    const form = component.find('input');
    form.props().onChange({target: {
      name: 'myName',
      value: 'myValue'
    }});
   
    expect(component.state('input')).toBeDefined();
  });

  it("should create an entry in component state with the event value", () => {
    const component = shallow(<UsersTest />);
    const form = component.find('input');
    form.props().onChange({target: {
      name: 'myName',
      value: 'myValue'
    }});
    
    expect(component.state('input')).toEqual('myValue');
  });
});