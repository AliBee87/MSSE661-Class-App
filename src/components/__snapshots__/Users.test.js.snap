// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users should render initial layout 1`] = `
Array [
  <div>
    <input
      onChange={[Function]}
      type="text"
    />
  </div>,
]
`;
